GroupProject/Items/clsItemsSQL.cs

```diff
@@ -1,13 +1,65 @@
-﻿using System;
+﻿using System.Text.RegularExpressions;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using GroupProject.Model;
 using GroupProject;
 
 namespace GroupProject.Items
 {
-    class clsItemsSQL
+    public class clsItemsSQL
     {
+        /// <summary>
+        /// This is a private data access dependency
+        /// </summary>
+        private clsDataAccess _dataAccess;
+
+        /// <summary>
+        /// Constructor to do DI for the clsDataAcess
+        /// </summary>
+        public clsItemsSQL(clsDataAccess dataAccess) 
+        {
+            _dataAccess = dataAccess;
+        }
+
+        /// <summary>
+        /// This is a method to update the jewelryItem that is changed when the save button is pressed
+        /// </summary>
+        public void UpdateJewelryItem(ItemViewModel itemViewModel) 
+        {
+            try
+            {
+                string sSql = $"UPDATE Item SET ItemDesc = {itemViewModel.Description}, ItemPrice = {itemViewModel.Price}, ItemName = {itemViewModel.Name} WHERE ItemNo = {itemViewModel.Code}";
+                int numOfRowsAffected = _dataAccess.ExecuteNonQuery(sSql);
+                Console.WriteLine($"The number of rows affected was {numOfRowsAffected}");
+
+            } 
+            catch (Exception ex)
+            {
+                throw new ApplicationException($"There was an issue with communicating with the database, trying to get {itemViewModel.Name}. Message: {ex.Message}");
```

Comment:
I think he wants the try/catch error messages to look a certain way, so instead use this for the catches in this file:
throw new Exception(MethodInfo.GetCurrentMethod().DeclaringType.Name + "." + MethodInfo.GetCurrentMethod().Name + " -> " + ex.Message);

Reply:
interesting... Well whatever. If that's what will get us the grade...

Reply:
These should be changed now